### PR TITLE
Fix test - file delete exception & add Github Actions

### PR DESCRIPTION
**Pull request by dansysanalyst**
Hi Wendell, 

Thank you so much for this great package and your amazing talk at PHP Lisbon.

This PR fixes the Exception when running tests:

```
• Tests\Feature\MakeDTOCommandTest > it validates that a new ValidatedDTO class is generated
   PHPUnit\Framework\ExceptionWrapper 

  unlink(/Users/dan/code/laravel-validated-dto/vendor/orchestra/testbench-core/laravel/app/DTOs/UserDTO.php): No such file or directory
```

## UPDATE

I would like to apologize to make so many changes in a single PR, I know it turns into a review nightmare, but I took your feedback and decided to go a bit further.

📚  I have added some badges to the README, including the supported Laravel versions. I think it would be a good idea to add it in the requirement sections in the doc.

---

📌  I have separated the tests into individual files to promote organization and easy access when maintaining. 

---

📌  Some tests were separated into two or more individual; I like to read the test output to check if I have all the cases and conditions covered

---

📌 I noticed that in some parts, the `$value` `type` was being checked and followed by a `toBe()`. I remove the redundancy since `toBe()` already verifies the content and type.

Example:

```
expect($array)
        ->toBeArray()
        ->toEqual(['key' => 'value']);
```

can be written as: 

```php
expect($array)->toBe(['key' => 'value']); 
```

---

📌 I have converted some tests to [Higher Order Tests](https://pestphp.com/docs/higher-order-tests). I believe it improves readability on simpler tests: 

```php
it('casts to integer', function () {
    $castable = new IntegerCast();

    $result = $castable->cast($this->testProperty, '5');
    expect($result)
        ->toBeInt()
        ->toBe(5);
});
```

Simplifies to:

```php
it('properly casts to integer')
    ->expect(fn () => new IntegerCast())
    ->cast(test_property(), '5')
    ->toBe(5);
```

---

📌 I noticed the usage of beforeEach() on many test files, and all of them returned the same string. 

I opted for moving it into a [Custom Helper](https://pestphp.com/docs/helpers) inside Pest.php, so all tests reference to the same single source, re-using code.

```php 
beforeEach(function () {
    $this->testProperty = 'test_property';
});
```

Changes to:

```
function test_property(): string
{
    return 'test_property';
}
```

and can simply be used as:

```php 
expect(test_property())->toBe('test_property');
```

---

📌 I have created a [Custom Expectation](https://pestphp.com/docs/expectations#custom-expectations) to check if a file exists and has a certain content.

It can be used as follows:

```php
test('a new file will be created', function () {
    file_put_contents('/tmp/dan', 'hello world');

    expect('/tmp/dan')->toBeFileWithContent('hello world');
});
```

I have some remarks on using it with the `not()` modifier as it could return an ambiguous result (file doesn't exist or file 
exist but without the specified content) but I think it improves readability in the artisan test readability.

Greetings and thanks,

Dan

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest]
+        php: [8.2, 8.1]
```
Comment (WendellAdriel) [inline .github/workflows/tests.yml:16]: Can you add 8.0 as well, please?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        laravel: [9.*]
```
Comment (WendellAdriel) [inline .github/workflows/tests.yml:17]: Is it possible to already test with 10.*?

**Review by WendellAdriel — CHANGES_REQUESTED** [review]
Amazing PR,thanks for the help!
I just added two small comments 😉

File: .github/workflows/tests.yml (hunk 1 above)
Comment (dansysanalyst) [reply .github/workflows/tests.yml]: ✅

File: .github/workflows/tests.yml (hunk 2 above)
Comment (dansysanalyst) [reply .github/workflows/tests.yml]: ✅

**Review by WendellAdriel — APPROVED** [review]
Wow 🤯 
That's what I call a refactor 💪🏼 
Amazing work here, @dansysanalyst 
Thank you so much 🥳

**Pull request merged**